xdsl/dialects/stablehlo.py

```diff
@@ -0,0 +1,41 @@
+from xdsl.dialects.builtin import TensorType
+from xdsl.ir import Attribute, Dialect, SSAValue
+from xdsl.irdl import IRDLOperation, irdl_op_definition, operand_def, result_def
+from xdsl.utils.exceptions import VerifyException
+
+
+@irdl_op_definition
+class AbsOp(IRDLOperation):
+    name = "stablehlo.abs"
+    operand = operand_def(TensorType)
+    result = result_def(TensorType)
+
+    def __init__(self, operand: SSAValue, result: Attribute):
+        super().__init__(operands=(operand,), result_types=(result,))
```

Comment:
We usually have the result_type be optional when it can be inferred from the operand:
```suggestion
    def __init__(self, operand: SSAValue, result_type: Attribute | None):
        if result_type is None:
            result_type = operand.type
        super().__init__(operands=(operand,), result_types=(result_type,))
```

Reply:
Just a note, I would prefer to keep the names as close as possible to the stablehlo specification. In the definition of `absOp`, `result` is used as opposed to `result_type`. So, if possible, I would rather keep the name.